app/scripts/modules/core/src/cancelModal/CancelModal.tsx

```diff
@@ -0,0 +1,133 @@
+import * as React from 'react';
+import { Modal } from 'react-bootstrap';
+import { Field, FieldProps, Form, Formik } from 'formik';
+import { IPromise } from 'angular';
+
+import { SubmitButton, ModalClose } from 'core';
+import { ReactModal } from 'core/presentation';
+
+export interface ICancelModalProps {
+  body?: string;
+  buttonText: string;
+  cancelButtonText?: string;
+  header: string;
+  submitMethod: (reason: string, force?: boolean) => IPromise<any>;
+  closeModal?(result?: any): void; // provided by ReactModal
+  dismissModal?(rejection?: any): void; // provided by ReactModal
+}
+
+export interface ICancelModalState {
+  isSubmitting: boolean;
+  error: boolean;
+  errorMessage?: string;
+}
+
+export interface ICancelModalValues {
+  reason?: string;
+}
+
+export class CancelModal extends React.Component<ICancelModalProps, ICancelModalState> {
+  public static defaultProps: Partial<ICancelModalProps> = {
+    cancelButtonText: 'Cancel',
+  };
+
+  public static confirm(props: ICancelModalProps): Promise<void> {
+    return ReactModal.show(CancelModal, props);
+  }
+
+  constructor(props: ICancelModalProps) {
```

Comment:
When you don't need to actually *do* anything in the constructor except set initial state, you can omit the constructor and just create a field.  I.e.:

```js
export class CancelModal {
  public state = { isSubmitting: false, error: false };
  ...
```

Note: what you have is fine, and we do it regularly, but I wanted to point out a different option you might not have known about

Reply:
I like that, much cleaner. Thanks.